### PR TITLE
[animations] Add RouteSettings to OpenContainer

### PR DESCRIPTION
**Pull request by georgeherby**
This adds RouteSettings to the OpenContainer so that this can be passed to the Navigator when doing Navigator.push.

Tested on my project and its working fine. I am not sure if there is a test that can be made to check the route setting is being set, then again this is just an override.

Related to issue:
https://github.com/flutter/flutter/issues/51399

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -4,3 +4,4 @@
 #   Name/Organization <email address>
 
 Google Inc.
+George Herbert <georgeherby@gmail.com>
```
Comment (goderbauer) [inline AUTHORS]: nit: please end file with a new line

File: packages/animations/lib/src/open_container.dart [hunk 2]
```diff
@@ -56,22 +56,23 @@ class OpenContainer extends StatefulWidget {
...
+      this.closedShape = const RoundedRectangleBorder(
+        borderRadius: BorderRadius.all(Radius.circular(4.0)),
+      ),
+      this.openShape = const RoundedRectangleBorder(),
+      @required this.closedBuilder,
+      @required this.openBuilder,
+      this.tappable = true,
+      this.transitionDuration = const Duration(milliseconds: 300),
+      this.transitionType = ContainerTransitionType.fade,
+      this.routeSettings})
```
Comment (goderbauer) [inline packages/animations/lib/src/open_container.dart]: nit: please add a `,` after routeSettings and reformat with dartfmt.

File: packages/animations/lib/src/open_container.dart [hunk 3]
```diff
@@ -222,19 +226,19 @@ class _OpenContainerState extends State<OpenContainer> {
...
+        openElevation: widget.openElevation,
+        closedShape: widget.closedShape,
+        openShape: widget.openShape,
+        closedBuilder: widget.closedBuilder,
+        openBuilder: widget.openBuilder,
+        hideableKey: _hideableKey,
+        closedBuilderKey: _closedBuilderKey,
+        transitionDuration: widget.transitionDuration,
+        transitionType: widget.transitionType,
+        settings: widget.routeSettings));
```
Comment (goderbauer) [inline packages/animations/lib/src/open_container.dart]: nit: please add a trailing comma after routeSettings and reformat with dartfmt.

**Comment by goderbauer** [comment]
@georgeherby Do you have plans to come back to this PR any time soon to address the feedback?

**Comment by georgeherby** [comment]
@goderbauer yes, it is on my todo list to get this looked at, not 100% sure how best to test it but i will have a look this weekend i hope

**Comment by goderbauer** [comment]
@georgeherby I am going to close this one for now. Feel free to reopen it or open a new one when you find the time to finish this PR.

**Pull request closed without merging**